chiselFrontend/src/main/scala/chisel3/core/Reg.scala

```diff
@@ -153,12 +153,15 @@ object RegInit {
       requireIsChiselType(t, "reg type")
     }
     val reg = t.cloneTypeFull
-    val clock = Builder.forcedClock.ref
-    val reset = Builder.forcedReset.ref
+    val clock = Builder.forcedClock
+    val reset = Builder.forcedReset
+    if (reset.isInstanceOf[AsyncReset] && init.litOption.isEmpty) {
+      Builder.error(s"Register $this with async reset has non-literal reset value")
```

Comment:
This has nothing to do with clock domains - it's just that async reset's default value is usually reset to a fixed constant. Logic synthesis tools can use SR flops to have non-constant async reset values but there are some implementation issues associated with that (e.g. glitches). Jack can substantiate

Reply:
I would add that we definitely need checks for legal clock domains (and after this PR, legal async reset deassert), but that's something we'll have to do in FIRRTL using https://github.com/freechipsproject/firrtl/pull/937

Reply:
Hey all! I have been messing around with the async-reset branch and have noticed that the litOption method (called in above diff) has been left unimplemented in the Aggregate class. This causes issues when trying to initialize a register of Vecs. I can come up with a local workaround for now, but I thought I would bring it to your attention in case you were unaware. I see the TODO is there so just tell me to shut up if this is already on your radar. All in all, very cool stuff!

Reply:
Yep, that's what #805 and #849 are meant to implement :)